### PR TITLE
Upgrade the project to Node 20

### PR DESCRIPTION
**Pull request by jonathanlinat**
This PR aims to upgrade the whole project to Node 20, as required here: https://github.com/TrenchBroom/TrenchBroom/pull/4467#issuecomment-1919982362

Once the PR is merged, a tag `0.8.0` must be created based on the latest changes pushed to the `master` branch.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
   "license": "MIT",
+  "engines": {
+    "bun": "please-use-npm",
+    "deno": "please-use-npm",
+    "node": ">=20 <21",
+    "npm": ">=10 <11",
+    "pnpm": "please-use-npm",
+    "yarn": "please-use-npm"
+  },
+  "packageManager": "npm@10.2.4",
```
Comment (jonathanlinat) [inline package.json:17]: This enforces the use of `Node 20` and `npm 10.x`.

File: build.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
-# install dependencies with yarn
-yarn install
+# install dependencies
+npm install
```
Comment (jonathanlinat) [inline build.sh:4]: We do not want to use another package manager. `npm` is enough for this project.

File: action.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ inputs:
     description: 'Token for the repository. This can be passed using {{ secrets.GITHUB_TOKEN }}'
     required: true
 runs:
-  using: 'node12'
+  using: 'node20'
```
Comment (jonathanlinat) [inline action.yml:8]: That's where the magic happens.

File: .npmrc [hunk 4]
```diff
@@ -0,0 +1 @@
+engine-strict=true
```
Comment (jonathanlinat) [inline .npmrc:1]: To work locally, we want to ensure the developer(s) have installed the required engine listed in the `package.json` file.

If not, the `npm` commands won't be executed, and a message will be sent to the developer inviting him/her to use `Node 20`.

**Comment by kduske** [comment]
Thank you!

**Pull request merged**